app/controllers/players_controller.rb

```diff
@@ -44,6 +44,19 @@ def update
     params=player_params
     unless is_organiser_view
       params[:user_id] = current_user.id
+      if @tournament.nrdb_deck_registration?
+        if @player.decks_locked?
```

Comment:
While setting a deadline for locking decks is something I think you should put in a follow-up PR, I’m not sure I would keep the locking stuff in here at all right now.

Reply:
I moved this to here and I'm making more changes: https://github.com/Project-NISEI/cobra/pull/145